code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -161,8 +161,11 @@
 
 /obj/item/tool/weldingtool/Destroy()
 	if(welding)
+		var/mob/user
 		if(ismob(loc))
-			loc.SetLuminosity(-2)
+			user = loc
+		if(loc && loc == user)
```

Comment:
This check is not necessary.
`if(loc)` is covered by `if(ismob(loc))`, so it will always be true when checked.
And `if(loc == user)` will always be true as a consequence, because `user = loc` will have happened.